index.js

```diff
@@ -8,35 +8,44 @@
 (function () {
 	'use strict';
 
-	var hasOwn = {}.hasOwnProperty;
-
-	function classNames () {
-		var classes = [];
```

Comment:
The main idea here is to go away from arrays.  
Due to the iterative process we can replace array accumulator with simple string.  
It gives us two benefits:
- String concatenation is faster than array push.
- Because we are generating the result on the fly - there is no need of `classes.join(' ')` which in fact caused second array iteration (first was over the function arguments).

Secondary: i've moved variables declaration out of the loops (it gives pretty big impact).
Plus, IE's optimizer works better if to make `typeof` separately without storing it's result into a variable.

Reply:
Could you quantify the performance differences of everything _other_ than the String concatenation change?
We have had nothing but to-and-fro changes between the two methods on this project.

The results appear to vary on every build of V8.

Reply:
@dcousens it impacts not only the V8 based browsers, FF has performance improvements too.